### PR TITLE
Geneva Exporter - Throw on TableNameMappings `null` value

### PR DESCRIPTION
**Pull request by mic-max**
Throw exception when null value is used in `TableNameMappings` for Geneva Exporter Options class.
Includes unit test.

### CODE REVIEW

**Review by reyang — APPROVED** [review]

**Comment by cijothomas** [comment]
@mic-max please address conflicts.

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs [hunk 1]
```diff
@@ -64,6 +65,8 @@ public GenevaLogExporter(GenevaExporterOptions options)
             var tempTableMappings = new Dictionary<string, string>(options.TableNameMappings.Count, StringComparer.Ordinal);
             foreach (var kv in options.TableNameMappings)
             {
+                Guard.ThrowIfNull(kv.Value);
```
Comment (TimothyMothra) [inline src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs:68]: I would recommend including the Key name in the exception message here.

Comment (utpilla) [reply src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs:68]: Would it be better to do this validation in the `GenevaExporterOptions` class, like how it's done for `PrepopulatedFields`: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Exporter.Geneva/GenevaExporterOptions.cs#L36-L73

**Comment by mic-max** [comment]
What should be done when a null key exists in the `TableNameMappings`, if anything?

File: src/OpenTelemetry.Exporter.Geneva/GenevaExporterOptions.cs [hunk 2]
```diff
@@ -27,11 +27,36 @@ public class GenevaExporterOptions
...
+    public IReadOnlyDictionary<string, string> TableNameMappings
+    {
+        get => this._tableNameMappings;
+        set
+        {
+            Guard.ThrowIfNull(value);
+
+            foreach (var entry in value)
+            {
+                if (entry.Value is null)
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.Geneva/GenevaExporterOptions.cs:45]: we can check if key is null/empty as well here.

Comment (mic-max) [reply src/OpenTelemetry.Exporter.Geneva/GenevaExporterOptions.cs:50]: Actually, C# already doesn't allow `null` to be a key in a Dictionary :)

File: src/OpenTelemetry.Exporter.Geneva/GenevaExporterOptions.cs [hunk 3]
```diff
@@ -27,11 +27,36 @@ public class GenevaExporterOptions
...
+                {
+                    throw new ArgumentNullException(entry.Key, $"{nameof(this.TableNameMappings)} must not contain null values.");
+                }
+            }
+
+            var copy = new Dictionary<string, string>(value.Count);
+            foreach (var entry in value)
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.Geneva/GenevaExporterOptions.cs:52]: we can validate and copy with a single foreach. There is no need of dual iteration here.

**Review by cijothomas — APPROVED** [review]

**Pull request merged**